service/controller/rbac_resource_set.go

```diff
@@ -57,16 +66,21 @@ func newTODOResourceSet(config todoResourceSetConfig) (*controller.ResourceSet,
 		}
 	}
 
-	// handlesFunc defines which objects you want to get into your controller, e.g. which objects you want to watch.
+	// filter namespaces without proper labels.
 	handlesFunc := func(obj interface{}) bool {
-		// TODO: By default this will handle all objects of the type your controller is watching.
-		// Your controller is watching a certain kubernetes type, so why do we need to check again?
-		// Because there might be a change in the object structure - e.g. the type `AWSConfig` object might have the field
-		// availabilityZones recently, but older ones don't, and you don't want to handle those.
-		//
-		// Normally we use this to filter objects containing the expected `versionbundle` version.
-		// So two versions of your operator don't accidentally reconcile the same CR.
-		return true
+		namespace, err := namespaceauth.ToNamespace(obj)
+		if err != nil {
+			return false
+		}
+
+		_, clusterLabelOK := namespace.ObjectMeta.Labels[nsClusterLabel]
+		_, orgLabelOK := namespace.ObjectMeta.Labels[nsOrgLabel]
```

Comment:
I think what we try to do here is to watch objects having a certain label so we should configure a label selector for the controller. See https://github.com/giantswarm/operatorkit/blob/master/controller/controller.go#L84-L85.

Reply:
I didn't know about this. Thank you :heart: